src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/SignalMethodProvider.cs

```diff
@@ -23,8 +25,9 @@ public SignalMethodProvider(WorkflowExecutionContext workflowContext, ISecurityT
                 {
                     var payload = !String.IsNullOrWhiteSpace(workflowContext.CorrelationId) ? SignalPayload.ForCorrelation(signal, workflowContext.CorrelationId) : SignalPayload.ForWorkflow(signal, workflowContext.WorkflowId);
                     var token = securityTokenService.CreateToken(payload, TimeSpan.FromDays(7));
-                    var urlHelper = serviceProvider.GetRequiredService<IUrlHelper>();
-                    return urlHelper.Action("Trigger", "HttpWorkflow", new { area = "OrchardCore.Workflows", token });
+                    var urlHelperFactory = serviceProvider.GetRequiredService<IUrlHelperFactory>();
```

Comment:
How about using LinkGenerator?

Reply:
Yes good idea, which uses internally the urlHelper, as I remember.

Reply:
@ns8482e done